wait_for_it/wait_for_it.py

```diff
@@ -54,7 +54,7 @@ async def _wait_until_available(host, port):
             if sys.version_info[:2] >= (3, 7):
                 await writer.wait_closed()
             break
-        except (socket.gaierror, ConnectionError):
+        except (socket.gaierror, ConnectionError, OSError):
```

Comment:
I just learned we'll get `TypeError` for service `[::1]:1234` here if the host doesn't have full IPv6 support :disappointed:

Reply:
I will add `TypeError` and revert `ConnectionRefusedError` back to `OSError` due to confirmation of failure: https://github.com/clarketm/wait-for-it/pull/26/checks?check_run_id=946336074

Reply:
Do you have any ideas on how to test this condition in CI?

Reply:
> Do you have any ideas on how to test this condition in CI?

Interesting question!

The last point can be reproduced on even a box with working IPv6 like this:

```console
$ python3.7 -c "import socket; socket.getaddrinfo(10, b'\x04\xd2\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', socket.AddressFamily.AF_INET6, socket.SocketKind.SOCK_STREAM, 6, 0)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
TypeError: getaddrinfo() argument 1 must be string or None
```
Notice how `host` is integer 10 and `port` is a bytes instance.

We could use `patch.object(..)` from `unittest.mock.patch` to make `socket.getaddrinfo` raise `TypeError` the same way it happens above. I'm unsure though if that's going to provide value. What do you think?

Reply:
Instead of patching directly, could we disable ipv6 on the host and assert that a `TypeError` is raised? That has some value IMO but I have not looked into it so it may be nontrivial.

Reply:
For Python, IPv6 support is toggled at compile time using switch `--disable-ipv6`. I am not really in favor of having the CI compile a non-IPv6 copy of Python at the moment.